app/models/pipeline_run.rb

```diff
@@ -153,38 +162,317 @@ def retry
   end
 
   def report_ready?
-    job_status == STATUS_CHECKED || (ready_step && active_stage && active_stage.step_number > ready_step)
+    clause_for_old_runs = result_status.nil? && (job_status == STATUS_CHECKED || (ready_step && pipeline_run_stages.find_by(step_number: ready_step) && pipeline_run_stages.find_by(step_number: ready_step).job_status == STATUS_LOADED))
+    # TODO: migrate old runs so we don't need to deal with them separately in the code
+    result_status_for(REPORT_READY_OUTPUT) == STATUS_LOADED || clause_for_old_runs
   end
 
   def succeeded?
     job_status == STATUS_CHECKED
   end
 
+  def db_load_ercc_counts
+    # TODO: break this function up into 3 different target_outputs / loader commands
+    #       rather than combining several outputs in 1 function
+
+    # Load version info applicable to host filtering
+    version_s3_path = "#{host_filter_output_s3_path}/#{VERSION_JSON_NAME}"
+    self.version = `aws s3 cp #{version_s3_path} -`
+
+    # Check if input was truncated
+    truncation_file = "#{host_filter_output_s3_path}/#{INPUT_TRUNCATED_FILE}"
+    _stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", truncation_file)
+    self.truncated = `aws s3 cp #{truncation_file} -`.split("\n")[0].to_i if status.exitstatus.zero?
+    save
```

Comment:
How does this save not clobber updates from other resque jobs that are trying to modify the same pipeline run?

Reply:
Fixed clobbering of result_status according to our discussion. The columns modified in db_load_ercc_counts should not get clobbered, because there is only one job executing this function for each pipeline_run.